### PR TITLE
autoconf: fix Linux build

### PR DESCRIPTION
**Pull request by barracuda156**
Closes: https://trac.macports.org/ticket/70673

#### Description

Non-macOS-only:

Use MacPorts perl5 on non-Darwin systems.
Also it may need `help2man`.

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

Linux riscv64

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by barracuda156** [comment]
If there are any objections, I do not insist on anything for Linux, but this does fix the issue for me.

File: devel/autoconf/Portfile [hunk 1]
```diff
@@ -44,9 +44,16 @@ post-patch {
...
-configure.perl      /usr/bin/perl
+if {${os.platform} eq "darwin"} {
+    # It should be safe to use the system Perl, since the scripts only use
+    # core modules. The README prescribes 5.6 or later, and Tiger has 5.8.6.
+    configure.perl  /usr/bin/perl
+} else {
+    depends_build-append \
+                    port:help2man
+    # https://trac.macports.org/ticket/70673
+    configure.perl  ${prefix}/bin/perl5
```
Comment (ryandesign) [inline devel/autoconf/Portfile:55]: help2man depends on perl5.34. perl5.34 provides the perl5.34 executable. The perl5 symlink is provided by the perl5 port. If you want to use the perl5 symlink here you have to depend on the perl5 port.

**Comment by reneeotten** [comment]
ping @barracuda156 - please address the comment by @ryandesign

**Comment by barracuda156** [comment]
@reneeotten @ryandesign Yeah, will do this tomorrow. Sorry for a delay, Linux is not really a priority, but I want basic ports to work.

**Comment by barracuda156** [comment]
@reneeotten Done, sorry for a delay here.

**Comment by reneeotten** [comment]
> @reneeotten Done, sorry for a delay here.

You mixed up two changes in the last commit and the commit message does not reflect that. The "perl5" dependency for "autoconf" should go in the first commit.

**Pull request merged**